internal/server/task_test.go

```diff
@@ -222,6 +227,61 @@ func TestPostTask(t *testing.T) {
 	assert.Equal(t, http.StatusOK, resp.StatusCode)
 }
 
+func TestPostTaskWhenAssignToMe(t *testing.T) {
+    user := sirius.User{ID: 66, DisplayName: "Me", Roles: []string{"OPG User", "Reduced Fees User"}}
+
+	client := &mockTaskClient{}
+	client.
+		On("TaskTypes", mock.Anything).
+		Return([]string{"a", "b"}, nil)
+	client.
+		On("Teams", mock.Anything).
+		Return([]sirius.Team{{ID: 1, DisplayName: "A Team"}}, nil)
+	client.
+		On("Case", mock.Anything, 123).
+		Return(sirius.Case{UID: "7000-0000-0000", CaseType: "LPA"}, nil)
+	client.
+		On("CreateTask", mock.Anything, 123, sirius.TaskRequest{
+			Type:        "Some task type",
+			DueDate:     "2022-03-04",
+			Name:        "Do this",
+			Description: "Please",
+			AssigneeID:  66,
+		})
```

Comment:
You'll need to include a `Return(...)` method here to explain what the expected return value is. This means you're saying to Go "when you call `CreateTask` with these arguments, I expect you to return `nil`"

```suggestion
		}).
		Return(nil)
```

That's needed because the [`CreateTask` method](https://github.com/ministryofjustice/opg-sirius-lpa-frontend/blob/c60df9661f29854a6a213f87db090eab4cb6c974/internal/sirius/create_task.go#L18) has a required return type of `error`. Since this test is mocking a scenario where the request is successful, you can just make it return `nil`, which is treated as "no error occurred".